pkg/cmd/server/api/types.go

```diff
@@ -194,6 +194,11 @@ type NodeConfig struct {
 	// system, this value should be set to the upstream nameservers dnsmasq resolves with.
 	DNSNameservers []string
 
+	// DNSUpstreamResolvConf is a path to a resolv.conf file that contains settings for an upstream server.
+	// Only the nameservers and port fields are used. The file must exist and parse correctly. It overrides
+	// nameservers in DNSNameservers if set.
+	DNSUpstreamResolvConf string
```

Comment:
It's the right word though - it's the resolv conf that we defer to.  DNSRecursiveResolvConf is another option I guess.